platforms/apple/build_xcframework.py

```diff
@@ -69,29 +21,82 @@ def get_framework_build_command_for_platform(platform, destination, framework_na
     parser.add_argument('--framework_name', default='opencv2', help='Name of OpenCV xcframework (default: opencv2, will change to OpenCV in future version)')
+    parser.add_argument('--iphoneos_archs', default=None, help='select iPhoneOS target ARCHS. Default is "armv7,arm64"')
+    parser.add_argument('--iphonesimulator_archs', default=None, help='select iPhoneSimulator target ARCHS. Default is "x86_64,arm64"')
+    parser.add_argument('--macos_archs', default=None, help='Select MacOS ARCHS. Default is "x86_64,arm64"')
+    parser.add_argument('--catalyst_archs', default=None, help='Select Catalyst ARCHS. Default is "x86_64,arm64"')
+    parser.add_argument('--build_only_specified_archs', default=False, action='store_true', help='if enabled, only directly specified archs are built and defaults are ignored')
     parser.add_argument('passthrough_args', nargs=argparse.REMAINDER, help='Any flags not captured by this script will be passed through to the ios/osx build_framework.py scripts')
 
     args, unknown_args = parser.parse_known_args()
-    print(f"The following args will be passed through to the ios/osx build_framework.py scripts: {args.passthrough_args}")
+    if args.passthrough_args:
+        print(f"The following args are not recognized by this script and will be passed through to the ios/osx build_framework.py scripts: {args.passthrough_args}")
+
+    # Parse architectures from args
+
+    iphoneos_archs = args.iphoneos_archs
+    if not iphoneos_archs and not args.build_only_specified_archs:
+        # Supply defaults
+        iphoneos_archs = "armv7,arm64"
+    print(f'Using iPhoneOS ARCHS={iphoneos_archs}')
+
+    iphonesimulator_archs = args.iphonesimulator_archs
+    if not iphonesimulator_archs and not args.build_only_specified_archs:
+        # Supply defaults
+        iphonesimulator_archs = "x86_64,arm64"
+    print(f'Using iPhoneSimulator ARCHS={iphonesimulator_archs}')
 
-    platforms = args.platforms.split(',')
+    macos_archs = args.macos_archs
+    if not macos_archs and not args.build_only_specified_archs:
+        # Supply defaults
+        macos_archs = "x86_64,arm64"
+    print(f'Using MacOS ARCHS={macos_archs}')
+
+    catalyst_archs = args.macos_archs
+    if not catalyst_archs and not args.build_only_specified_archs:
+        # Supply defaults
+        catalyst_archs = "x86_64,arm64"
+    print(f'Using Catalyst ARCHS={catalyst_archs}')
+
+    # Build phase
 
     try:
         # Build .frameworks for each platform
+        osx_script_path = os.path.abspath(os.path.abspath(os.path.dirname(__file__))+'/../osx/build_framework.py')
+        ios_script_path = os.path.abspath(os.path.abspath(os.path.dirname(__file__))+'/../ios/build_framework.py')
+
         build_folders = []
-        for platform in platforms:
-            folder = f"./{args.out}/{platform}"
-            pathlib.Path(folder).mkdir(parents=True, exist_ok=True)
```

Comment:
@colejd I think actually we still want this, and should possibly infer some form of `platform` from the archs. I'm having issues with `OUTDIR` at the moment since this commit. Previous I was able to do something like:

```
$ python platforms/apple/build_xcframework.py --dynamic build/dynamic-xcframework
```

and then simultaneously in another terminal run

```
$ python platforms/apple/build_xcframework.py build/static-xcframework
```

This is no longer the case, and both builds seem to end up in the same new folder

```
opencv2/build/build-armv7-iphoneos
```

What I want is for them to end up here:

```
build/static-xcframework/build/build-armv7-iphoneos
build/dynamic-xcframework/build/build-armv7-iphoneos
```

Reply:
@colejd Ah found the issue: https://github.com/Rightpoint/opencv/pull/4/commits/aa0b28fae2e1350738bca2bfc072b62165f913c2